packages/app/src/lib/import.ts

```diff
@@ -112,6 +118,29 @@ export async function asCSV(file: File, nameColIndex?: number, tagsColIndex?: nu
     return fromTable(parsed.data, nameColIndex, tagsColIndex);
 }
 
+export async function asCSVColumns(file: File): Promise<ImportCSVColumn[]> {
+    const data = await readFileAsText(file);
+    const papa = await loadPapa();
+    const parsed = papa.parse(data);
+    if (parsed.errors.length) {
+        throw new Err(ErrorCode.INVALID_CSV, "Failed to parse .csv file.");
+    }
+    const rows = parsed.data as string[][];
+
+    const columnNames = rows[0].map((column) => column.toLowerCase());
+
+    return columnNames.map((columnName) => {
+        const type = (Object.keys(FIELD_DEFS).filter((fieldType) => columnName.includes(fieldType))[0] ||
```

Comment:
I initially tried to use the `FIELD_DEFS.*.pattern` for matching, but many of them are `.*` so I didn't get a good "weight" for them anyway. This seems good enough, and we can later add "similarNames" or something like that to `FIELD_DEFS.*`, and use here as matching options. Could also help with #223 (which this kind of solves already)

Reply:
Thanks, that's helpful!